File: abc112/c.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <cmath>
#include <cstdio>
#include <numeric>
#include <vector>
#include <queue>
#include <map>
typedef unsigned long long ull;
typedef long long ll;
#define MAX 101
using namespace std;

struct point {int x,y,h;};
point C[MAX];
int main() {
  int n;
  cin >> n;
  for(int i=0;i<n;i++) {
    cin >> C[i].x >> C[i].y >> C[i].h;
  }
  point num;
  for(int x=0;x<=100;x++) {
    int h;
    for(int y=0;y<=100;y++) {
      bool flag=true;
      for(int i=1;i<n;i++) {
        h = abs(x-C[i-1].x)+abs(y-C[i-1].y)+C[i-1].h;
        if(C[i-1].h == 0) {
          flag=false;
          break;
        }
        if(h == abs(x-C[i].x)+abs(y-C[i].y)+C[i].h) {
          h=abs(x-C[i].x)+abs(y-C[i].y)+C[i].h;
        } else {
          flag=false;
          break;
        }
        if(flag) {
          num.x=x;
          num.y=y;
          num.h=h;
          break;
        }
      }
    }
  }
  cout << num.x << " " << num.y << " " << num.h << endl;
  return 0;
}

File: abc112/c.py
N = int(input())
X = []
Y = []
hh = []
vec = list()
G = (-1,-1,-1)

for _ in range(N):
    x, y, h = map(int, input().split())
    X.append(x)
    Y.append(y)
    hh.append(h)
    # 高さが1以上の三組を一つ取る(この場合, hi = H - |Xi - Cx| - |Yi - Cy|が成立する)
    if h >= 1:
        G = (x, y, h)

for i in range(0,101):
    for j in range(0,101):
        # 最初にHを固定する
        H = G[2] + abs(G[0] - i) + abs(G[1] - j)
        H = max(H, 0)
        flag = True
        for k in range(N):
            key = H - abs(X[k] - i) - abs(Y[k] - j)
            key = max(key, 0)
            if key != hh[k]:
                flag = False
        # 全てのkについてHが成立するとき, append
        if flag:
            vec.append((i,j,H))

if len(vec) == 1:
    print(vec[0][0], vec[0][1], vec[0][2])
